calc.rb

```diff
@@ -0,0 +1,101 @@
+# Elsa's calculator project
+# Due: 150508
+# Breakfast + Lunch + Dinner
+
+# welcome message, instructions, and receiving user input
+print "\nWelcome to the Calculator!\n\nYou will be asked to enter\n  1) your first number,\n  2) your mathematical operation,\n  3) and your second number.\nThe resulting calculation will then be provided.\n"
+
+# sets up while loop for repeated calculations
+restart = "Y"
+while restart == "Y" do
+
+
+  # asks for the first number
+  print "\nFirst number:\t"
+  first_number = gets.chomp
+  # if not a number, asks again
+  while first_number != "0" && first_number.to_i == 0 do
+    puts "\nNice try. How about a number this time?"
+    print "First number:\t"
+    first_number = gets.chomp
+  end
+
+
+  # asks for mathematical operation (without immediate  validation *sad face*)
+  print "Math operation:\t"
+  operation = gets.chomp.downcase
+
+  # asks for a second number
+  print "Second number: \t"
+  second_number = gets.chomp
+  # if not a number, asks again
+  while second_number != "0" && second_number.to_i == 0 do
+    puts "\nNice try. How about a number this time?"
+    print "Second number:\t"
+    second_number = gets.chomp
+  end
+
+
+
+  # operator handling and result generator
+  case operation
+
+  # addition
+  when "+", "plus", "and", "add", "addition"
```

Comment:
So I like you're using case to handle the conditionals. I think the limiting thing is that doing it this way looks like it impedes your ability to validate the operator when it's first entered, right?

(This is a little advanced, so don't worry not knowing about it, but keep it in mind in the future).

```
addition = [ "+", "plus", "and", "add", "addition" ]

when *addition
  result = first_number.to_i + second_number.to_i
  puts "#{first_number.to_i} + #{second_number.to_i} = #{result}"
...
```
The splat operator (`*`) unrolls an array so you can feed it into stuff like `when`, which accepts a sequence, but not an array.

So if you did it this way, you could declare a list of all the operators, and you would be able to check for them on input like it looks like you wanted. No more sadface!